asllib/doc/ASLTypingReference.tex

```diff
@@ -8694,2385 +8629,4792 @@ \subsection{Comments}
+    \item define $\newenvandfs$ as the list with \head\ $(\lenv, \vfone)$ and \tail\ $\envandfstwo$.
+  \end{itemize}
+\end{itemize}
+
+\begin{emptyformal}
+\subsection{Formally}
+\begin{mathpar}
+\inferrule[empty]{}{
+  \foldenvandfs(\genv, \overname{\emptylist}{\envandfs}) \typearrow (\overname{\genv}{\newgenv}, \overname{\emptylist}{\newenvandfs})
+}
+\and
+\inferrule[non\_empty]{
+  \tenv \eqdef (\genv, \lenv)\\
+  \declareonefunc(\tenv, \vf) \typearrow (\tenvone, \vfone) \OrTypeError\\\\
+  \foldenvandfs(G^\tenvone, \envandfsone) \typearrow (\newgenv, \envandfstwo) \OrTypeError\\\\
+  \newenvandfs \eqdef [(\lenv, \vfone)] \concat \envandfstwo
+}{
+  \foldenvandfs(\genv, \overname{[(\lenv, \vf)]\concat\envandfsone}{\envandfs}) \typearrow (\overname{\genv}{\newgenv}, \newenvandfs)
+}
+\end{mathpar}
+\end{emptyformal}
+
+\section{TypingRule.DefDecl \label{sec:TypingRule.DefDecl}}
+\hypertarget{def-defdecl}{}
+The function
+\[
+\defdecl(\overname{\decl}{\vd}) \aslto \overname{\identifier}{\name}
+\]
+returns the identifier $\name$ being defined by the declaration $\vd$.
+
+\subsection{Prose}
+One of the following applies:
+\begin{itemize}
+  \item All of the following apply (\textsc{d\_func}):
+  \begin{itemize}
+    \item $\vd$ declares a subprogram for the identifier $\name$.
+  \end{itemize}
+
+  \item All of the following apply (\textsc{d\_globalstorage}):
+  \begin{itemize}
+    \item $\vd$ declares a global storage element for the identifier $\name$.
+  \end{itemize}
+
+  \item All of the following apply (\textsc{d\_typedecl}):
+  \begin{itemize}
+    \item $\vd$ declares a type for the identifier $\name$.
+  \end{itemize}
+\end{itemize}
+
+\begin{emptyformal}
+\begin{mathpar}
+\inferrule[d\_func]{}{
+  \defdecl(\overname{\DFunc(\funcname: \name, \ldots)}{\vd}) \typearrow \name
+}
+\and
+\inferrule[d\_globalstorage]{}{
+  \defdecl(\overname{\DGlobalStorage(\GDname: \name, \ldots)}{\vd}) \typearrow \name
+}
+\and
+\inferrule[d\_typedecl]{}{
+  \defdecl(\overname{\DTypeDecl(\name, \Ignore, \Ignore)}{\vd}) \typearrow \name
+}
+\end{mathpar}
+\end{emptyformal}
+
+\section{TypingRule.DefEnumLabels \label{sec:TypingRule.DefEnumLabels}}
+\hypertarget{def-defenumlabels}{}
+The function
+\[
+\defenumlabels(\overname{\decl}{\vd}) \aslto \overname{\pow{\identifier}}{\vlabels}
+\]
+takes a declaration $\vd$ and returns the set of enumeration labels it defines --- $\vlabels$ --
+if it defines any.
+
+\subsection{Prose}
+One of the following applies:
+\begin{itemize}
+  \item All of the following apply (\textsc{decl\_enum}):
+  \begin{itemize}
+    \item $\vd$ is a declaration of an enumeration type with labels $\vlabels$;
+    \item the result is $\vlabels$ as a set (rather than a list).
+  \end{itemize}
+
+  \item All of the following apply (\textsc{other}):
+  \begin{itemize}
+    \item $\vd$ is not a declaration of an enumeration type;
+    \item define $\vlabels$ as the empty set.
```

Comment:
vlabel is the empty set

Reply:
I am trying to distinguish between different types of equality statements. This is done in the mathematical part but not in the Prose. Rich also noted on the confusion this causes.
So I am trying to move the Prose to use "define x as y" to mean I am defining an alias and "x is something" as an equality assertion.